lcls-twincat-pmps/PMPS/HelperFunctions/F_SafeBPCompare0Rate.TcPOU

```diff
@@ -3,24 +3,46 @@
   <POU Name="F_SafeBPCompare0Rate" Id="{61333098-b080-43b8-8c0d-b267fc83ae7a}" SpecialFunc="None">
     <Declaration><![CDATA[(*Compares BeamParam1 to BeamParam2: if the parameters of BeamParam1 are more conservative than BeamParam2
 the result will be true. *)
-(* 0-rate means this function will return true regardless of other conditions if the beam rate
+(* When the Mode PV is in Cu 0-rate means this function will return true regardless of other conditions if the beam rate
+is zero *)
+(* When the Mode PV is in SC 0-bc means this function will return true regardless of other conditions if the beam class
 is zero *)
 FUNCTION F_SafeBPCompare0Rate : BOOL
 VAR_INPUT
     BeamParam1    :    ST_BeamParams; //Must be more conservative than 2
     BeamParam2    :    ST_BeamParams;
 END_VAR
 VAR
-    // Beam-rate is zero, masks all others considerations.
+    {attribute 'pytmc' := '
+        pv: LFE:LCLSMODE
+        link: SIOC:FEEH:MP01:LCLSMODE_RBV
+    '}
+    HXR_MODE : FB_LREALFromEPICS;
+	{attribute 'pytmc' := '
+        pv: KFE:LCLSMODE
+        link: SIOC:FEES:MP01:LCLSMODE_RBV
```

Comment:
Question: how does this behave if the PV here doesn't connect?

Reply:
Should Fault the beam